src/Microsoft.AspNet.Authentication.Cookies/Events/BaseCookieContext.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Claims;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Authentication;
+
+namespace Microsoft.AspNet.Authentication.Cookies
+{
+    public class BaseCookieContext : BaseContext
```

Comment:
We nuked Options/Generics off of the base contexts, and just added them all back in the specific OAuth/Cookies/Facebook contexts themselves.  Let us get rid of the TOption generics on the base context classes

```
        public CookieAuthenticationOptions Options { get; }
```